### ISSUE TITLE
indent bug with multi-line arrays

### ISSUE THREAD

**Issue by woollybogger** [eslint/eslint#4203]
Code:

```js
/* eslint indent: [2, 2, {"SwitchCase": 1}] */
[
  'b',
  'a',
].sort().should.deepEqual([
  'a',
  'b',
]);
```

ESLint Output:

```cpp
2:1  error  Expected indentation of 2 space characters but found 0  indent
3:3  error  Expected indentation of 4 space characters but found 2  indent
4:3  error  Expected indentation of 4 space characters but found 2  indent
5:1  error  Expected indentation of 2 space characters but found 0  indent
```

Then, if I actually indent lines 2-5 by 2 spaces, ESLint says this:

```cpp
2:3  error  Expected indentation of 4 space characters but found 2  indent
3:5  error  Expected indentation of 6 space characters but found 4  indent
4:5  error  Expected indentation of 6 space characters but found 4  indent
5:3  error  Expected indentation of 4 space characters but found 2  indent
6:3  error  Expected indentation of 4 space characters but found 2  indent
7:3  error  Expected indentation of 4 space characters but found 2  indent
8:1  error  Expected indentation of 2 space characters but found 0  indent
```

The indentation of that code should not cause an error.

Originally found when using ESLint here: https://github.com/woollybogger/string-natural-compare/blob/master/test/test.js#L177

Using ESLint version 1.7.1

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by woollybogger** [comment]
Looks like this is related to #4161
